storage/s3_test.go

```diff
@@ -398,3 +406,180 @@ func TestS3Retry(t *testing.T) {
 		})
 	}
 }
+
+// Credit to aws-sdk-go
+func val(i interface{}, s string) interface{} {
+	v, err := awsutil.ValuesAtPath(i, s)
+	if err != nil || len(v) == 0 {
+		return nil
+	}
+	if _, ok := v[0].(io.Reader); ok {
+		return v[0]
+	}
+
+	if rv := reflect.ValueOf(v[0]); rv.Kind() == reflect.Ptr {
+		return rv.Elem().Interface()
+	}
+
+	return v[0]
+}
+
+func TestS3CopyEncryptionRequest(t *testing.T) {
```

Comment:
this test is identical to the `acl` test. please take a look how we could merge them.

Reply:
since acl and kms related changes are in two different pull requests, should i merge these two PRs or should we merge after both get into master branch?

Reply:
another pr is ok